### PR TITLE
[Fix] adding validation message for wrong upload type

### LINKED ISSUE #13000: [Bug]: Field Type Image Allowed PDF, txt etc... files
### Expected behavior

There should be a validation message.

### Actual behavior

When we use image type we can upload only images, here we can not use other files on object but when we select PDF or any other extension file into the image type field and is uploaded into the assets.

### Steps to reproduce

1. Add an image type field **[Upload Path: /Rushit]** into the class.
2. Create the object and select PDF, txt, or any other file instead of an image(png, jpeg, etc..).

->> Output is like the file that is not viewed into an object but we can see this file in the Assets.
image

### PR DESCRIPTION
**Pull request by mattamon**
## Changes in this pull request
Adding validation message for wrong upload type
Resolves #13000 

## Additional info  
See #13000 for description

### CODE REVIEW

File: bundles/AdminBundle/Controller/Admin/Asset/AssetController.php [hunk 1]
```diff
@@ -527,6 +527,17 @@ protected function addAsset(Request $request, Config $config)
...
         }
 
+        // check if there is a requested type and if matches the asset type of the uploaded file
+        $type = $request->get('type');
+        if($type) {
+            $mimetype = MimeTypes::getDefault()->guessMimeType($sourcePath);
+            $assetType = Asset::getTypeFromMimeMapping($mimetype, $filename);
+
+            if($type !== $assetType) {
+                throw new \Exception("Mime type does not match asset type: $type");
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/Asset/AssetController.php]:
```suggestion
                throw new \Exception("Mime type does not match with asset type: $type");
```

**Review by dvesh3 — CHANGES_REQUESTED** [review]
It works like a charm on DataObject Image Data type. We should also consider this for document image editable:
https://github.com/pimcore/pimcore/blob/5bf7298f7a7601da7839e647831030a5f656fbd5/bundles/AdminBundle/Resources/public/js/pimcore/document/editables/image.js#L231

File: bundles/AdminBundle/Resources/public/js/pimcore/document/editables/image.js [hunk 2]
```diff
@@ -242,7 +244,17 @@ pimcore.document.editables.image = Class.create(pimcore.document.editable, {
...
+        function (res) {
+            const response = Ext.decode(res.response.responseText);
+            if (response && response.success === false) {
+                pimcore.helpers.showNotification(t("error"), response.message, "error",
+                    res.response.responseText);
+            } else {
+                pimcore.helpers.showNotification(t("error"), res, "error",
+                    res.response.responseText);
+            }
+        }.bind(this), [] ,this.type);
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/public/js/pimcore/document/editables/image.js]:
```suggestion
        }.bind(this), [] ,this.getType());
```
we can kick the newly introduced property 😉

File: bundles/AdminBundle/Resources/public/js/pimcore/document/editables/image.js [hunk 3]
```diff
@@ -14,6 +14,8 @@
 pimcore.registerNS("pimcore.document.editables.image");
 pimcore.document.editables.image = Class.create(pimcore.document.editable, {
 
+    type: "image",
+
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/public/js/pimcore/document/editables/image.js]:
```suggestion
```

File: bundles/AdminBundle/Resources/public/js/pimcore/document/editables/image.js (hunk 2 above)
Comment (mattamon) [reply bundles/AdminBundle/Resources/public/js/pimcore/document/editables/image.js]: Ah yes, I wanted to do it exactly like in the tag/image.js
Shouldn't we stick to one pattern? We could also use this property for the getType method?

Comment (dvesh3) [reply bundles/AdminBundle/Resources/public/js/pimcore/document/editables/image.js]: yes, I understand you tried to follow one pattern but then we have to introduce this property for abstract and other editables which requires more refactoring. we can just use the existing method here with less code changes.

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**